sys-fs/tmsu/tmsu-0.6.1.ebuild

```diff
@@ -0,0 +1,40 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit golang-build
+
+EGO_PN="github.com/oniony/TMSU/"
+DESCRIPTION="Files tagger and virtual tag-based filesystem"
+HOMEPAGE="https://github.com/oniony/TMSU/wiki"
+SRC_URI="https://github.com/oniony/TMSU/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="test zsh-completion"
+
+RDEPEND=""
+DEPEND="
+	dev-go/go-sqlite3
+	dev-lang/go
+	dev-libs/go-fuse
+	zsh-completion? ( app-shells/zsh )
+"
+
+src_unpack() {
+	default
+	mv TMSU-${PV} ${P} || die "Failed to move sorce directory."
```

Comment:
I get the following error if I just set S="${WORKDIR}/TMSU-${PV}"

```
>>> Creating Manifest for /home/chymera/src/gentoo/sys-fs/tmsu
 * tmsu-0.6.1.tar.gz SHA256 SHA512 WHIRLPOOL size ;-) ...                   [ ok ]
>>> Unpacking source...
>>> Unpacking tmsu-0.6.1.tar.gz to /var/tmp/portage/sys-fs/tmsu-0.6.1/work
>>> Source unpacked in /var/tmp/portage/sys-fs/tmsu-0.6.1/work
>>> Preparing source in /var/tmp/portage/sys-fs/tmsu-0.6.1/work/TMSU-0.6.1 ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/sys-fs/tmsu-0.6.1/work/TMSU-0.6.1 ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/sys-fs/tmsu-0.6.1/work/TMSU-0.6.1 ...
env GOPATH=/var/tmp/portage/sys-fs/tmsu-0.6.1/work/tmsu-0.6.1:/usr/lib/go-gentoo go build -v -work -x github.com/oniony/TMSU/
WORK=/var/tmp/portage/sys-fs/tmsu-0.6.1/temp/go-build778023660
can't load package: package github.com/oniony/TMSU: cannot find package "github.com/oniony/TMSU" in any of:
	/usr/lib/go/src/github.com/oniony/TMSU (from $GOROOT)
	/var/tmp/portage/sys-fs/tmsu-0.6.1/work/tmsu-0.6.1/src/github.com/oniony/TMSU (from $GOPATH)
	/usr/lib/go-gentoo/src/github.com/oniony/TMSU
 * ERROR: sys-fs/tmsu-0.6.1::gentoo failed (compile phase):
 *   (no error message)
 * 
 * Call stack:
 *     ebuild.sh, line 115:  Called src_compile
 *   environment, line 383:  Called golang-build_src_compile
 *   environment, line 339:  Called die
 * The specific snippet of code:
 *       "$@" || die
 * 
 * If you need support, post the output of `emerge --info '=sys-fs/tmsu-0.6.1::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=sys-fs/tmsu-0.6.1::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/sys-fs/tmsu-0.6.1/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sys-fs/tmsu-0.6.1/temp/environment'.
 * Working directory: '/var/tmp/portage/sys-fs/tmsu-0.6.1/work/TMSU-0.6.1'
 * S: '/var/tmp/portage/sys-fs/tmsu-0.6.1/work/TMSU-0.6.1'

 * Messages for package sys-fs/tmsu-0.6.1:

 * ERROR: sys-fs/tmsu-0.6.1::gentoo failed (compile phase):
 *   (no error message)
 * 
 * Call stack:
 *     ebuild.sh, line 115:  Called src_compile
 *   environment, line 383:  Called golang-build_src_compile
 *   environment, line 339:  Called die
 * The specific snippet of code:
 *       "$@" || die
 * 
 * If you need support, post the output of `emerge --info '=sys-fs/tmsu-0.6.1::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=sys-fs/tmsu-0.6.1::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/sys-fs/tmsu-0.6.1/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sys-fs/tmsu-0.6.1/temp/environment'.
 * Working directory: '/var/tmp/portage/sys-fs/tmsu-0.6.1/work/TMSU-0.6.1'
 * S: '/var/tmp/portage/sys-fs/tmsu-0.6.1/work/TMSU-0.6.1'
```

I think that some part of the golang-build class gets confused if the package is not under the expected path. This approach was suggested by @gktrk